bin/vmware-host-health.rb

```diff
@@ -0,0 +1,185 @@
+         short: '-d',
+         long: '--dryrun',
+         description: 'run without sending to sensu but see what would be sent',
+         boolean: true,
+         default: false
+
+  option :handlers,
+         description: 'Comma separated list of handlers',
+         long: '--handlers <HANDLER>',
+         proc: proc { |s| s.split(',') },
+         default: []
+
+  option :help,
+         short: '-h',
+         long: '--help',
+         description: 'Show this message',
+         on: :tail,
+         boolean: true,
+         show_options: true,
+         exit: 0
+
+  def send_client_socket(data)
+    if config[:dryrun]
+      puts data.inspect
+    else
+      sock = UDPSocket.new
+      sock.send(data + "\n", 0, '127.0.0.1', 3030)
+    end
+  end
+
+  def send_ok(check_name, msg, source)
+    event = {
+      'name' => check_name,
+      'source' => source,
+      'status' => 0,
+      'output' => "#{self.class.name} OK: #{msg}",
+      'handlers' => config[:handlers]
+    }
+    send_client_socket(event.to_json)
+  end
+
+  def send_warning(check_name, msg, source)
+    event = {
+      'name' => check_name,
+      'source' => source,
+      'status' => 1,
+      'output' => "#{self.class.name} WARNING: #{msg}",
+      'handlers' => config[:handlers],
+      'occurrences' => 2
+    }
+    send_client_socket(event.to_json)
+  end
+
+  def send_critical(check_name, msg, source)
+    event = {
+      'name' => check_name,
+      'source' => source,
+      'status' => 2,
+      'output' => "#{self.class.name} CRITICAL: #{msg}",
+      'handlers' => config[:handlers],
+      'occurrences' => 2
+    }
+    send_client_socket(event.to_json)
+  end
+
+  def send_unknown(check_name, msg)
+    event = {
+      'name' => check_name,
+      'source' => source,
+      'status' => 3,
+      'output' => "#{self.class.name} UNKNOWN: #{msg}",
+      'handlers' => config[:handlers]
+    }
+    send_client_socket(event.to_json)
+  end
+
+  def cache_file(data)
+    file = File.open('/var/log/sensu/vmware-plugin.json', 'w')
+    file.puts data
+    file.close
+  end
+
+  def read_file
+    if File.file?('/var/log/sensu/vmware-plugin.json')
+      file = File.read('/var/log/sensu/vmware-plugin.json')
+      json = JSON.parse(file)
+      return json
+    end
+    {}
+  end
+
+  def run
+    readcache = read_file
+    cachewrite = {}
+    u = URI.parse(config[:host])
+    vim = RbVmomi::VIM.connect host: u.host, user: config[:username], password: config[:password], insecure: config[:verify_ssl]
+    rootfolder = vim.serviceInstance.content.rootFolder
+    dc = rootfolder.childEntity
+    dc.each do |d|
+      d.hostFolder.childEntity.each do |h|
+        h.host.each do |hs|
+          if hs.summary.overallStatus == 'red'
+            hs.triggeredAlarmState.each do |a|
+              msg = "#{hs.name} currently has an alarm #{hs.summary.overallStatus} - #{a.alarm.info.name}"
+              cachewrite["#{hs.name.gsub!(/[^0-9A-Za-z]/, '')}_Alarm"] = hs.name
```

Comment:
im fine with doing a function. This will save from having to do a ! as it was more to save on lines. Im always replacing the same variable so it seemed fine but I do get your point.